pennylane_lightning/lightning_kokkos/lightning_kokkos.py

```diff
@@ -40,188 +38,34 @@
-    res = simulate(circuit, state)
-    jac = LightningKokkosAdjointJacobian(state, batch_obs=batch_obs).calculate_jacobian(circuit)
-    return res, jac
-
-
-def vjp(
-    circuit: QuantumTape,
-    cotangents: Tuple[Number],
-    state: LightningKokkosStateVector,
-    batch_obs=False,
-    wire_map=None,
-):
-    """Compute the Vector-Jacobian Product (VJP) for a single quantum script.
-    Args:
-        circuit (QuantumTape): The single circuit to simulate
-        cotangents (Tuple[Number, Tuple[Number]]): Gradient-output vector. Must
-            have shape matching the output shape of the corresponding circuit. If
-            the circuit has a single output, ``cotangents`` may be a single number,
-            not an iterable of numbers.
-        state (LightningKokkosStateVector): handle to the Lightning state vector
-        batch_obs (bool): Determine whether we process observables in parallel when
-            computing the VJP. This value is only relevant when the lightning
-            kokkos is built with OpenMP.
-        wire_map (Optional[dict]): a map from wire labels to simulation indices
-
-    Returns:
-        TensorLike: The VJP of the quantum script
-    """
-    if wire_map is not None:
-        [circuit], _ = qml.map_wires(circuit, wire_map)
-    state.reset_state()
-    final_state = state.get_final_state(circuit)
-    return LightningKokkosAdjointJacobian(final_state, batch_obs=batch_obs).calculate_vjp(
-        circuit, cotangents
-    )
-
-
-def simulate_and_vjp(
-    circuit: QuantumTape,
-    cotangents: Tuple[Number],
-    state: LightningKokkosStateVector,
-    batch_obs=False,
-    wire_map=None,
-):
-    """Simulate a single quantum script and compute its Vector-Jacobian Product (VJP).
-    Args:
-        circuit (QuantumTape): The single circuit to simulate
-        cotangents (Tuple[Number, Tuple[Number]]): Gradient-output vector. Must
-            have shape matching the output shape of the corresponding circuit. If
-            the circuit has a single output, ``cotangents`` may be a single number,
-            not an iterable of numbers.
-        state (LightningKokkosStateVector): handle to the Lightning state vector
-        batch_obs (bool): Determine whether we process observables in parallel when
-            computing the jacobian. This value is only relevant when the lightning
-            kokkos is built with OpenMP.
-        wire_map (Optional[dict]): a map from wire labels to simulation indices
-
-    Returns:
-        Tuple[TensorLike]: The results of the simulation and the calculated VJP
-    Note that this function can return measurements for non-commuting observables simultaneously.
-    """
-    if wire_map is not None:
-        [circuit], _ = qml.map_wires(circuit, wire_map)
-    res = simulate(circuit, state)
-    _vjp = LightningKokkosAdjointJacobian(state, batch_obs=batch_obs).calculate_vjp(
-        circuit, cotangents
-    )
-    return res, _vjp
-
```

Comment:
All these methods were moved into the `LightningBase` class to improve reusability.